src/operations/blas3/gemm_interleaved.hpp

```diff
@@ -486,28 +493,41 @@ class Gemm<input_t, output_t, /* DoubleBuffer = */ false, /* NbcA = */ false,
    * @param reg_res  2D register array used to store the result C
    */
   PORTBLAS_INLINE void compute_block(packet_type *reg_a, packet_type *reg_b,
-                                     packet_type *reg_res) noexcept {
+                                     packet_out_type *reg_res) noexcept {
 #pragma unroll
     for (int i = 0; i < item_cols; ++i) {
 #pragma unroll
       for (int j = 0; j < item_rows; ++j) {
 #pragma unroll
         for (int b = 0; b < item_batchs / VectorSize; ++b) {
 #ifdef __ADAPTIVECPP__
-          if constexpr (is_half<element_t>::value) {
-            *reg_res = reg_a[j * (item_batchs / VectorSize) + b] *
-                           reg_b[i * (item_batchs / VectorSize) + b] +
-                       *reg_res;
+          if constexpr (is_half<element_in_t>::value ||
+                        !std::is_same_v<element_in_t, element_out_t>) {
+#pragma unroll
+            for (int v = 0; v < VectorSize; ++v) {
```

Comment:
Can you elaborate why this loop is necessary?

Reply:
@s-Nick AdaptiveCpp doesn't support sycl::vec<sycl::half> for mad operation, so had to break it down to its elements through the loop